src/main/scala/viper/silver/plugin/standard/termination/transformation/FunctionCheck.scala

```diff
@@ -125,7 +126,65 @@ trait FunctionCheck extends ProgramManager with DecreasesCheck with ExpTransform
         }
       } else Nil
     }
-    proofMethods
+    if (respectFuncPermAmounts)
+      proofMethods
+    else
+      proofMethods.map(removeConcretePermissionAmounts)
+  }
+
+  /**
+    * Given a method, removed all concrete permission amounts and replaces them with wildcard if they are positive,
+    * otherwise with none.
+    */
+  def removeConcretePermissionAmounts[N <: Node](n: N): N = n.transform{
```

Comment:
Is there an assumption that this transformation is only valid if no other methods are called from the current method?

I would say there isn't, given that any call to a method with a fixed perm amount here should fail (as we cannot be sure we have enough permissions). However, in the presence of permission introspection, I think it is possible to again reinstate reasoning about specific perm amounts